quickwit/quickwit-metastore/src/metastore/index_metadata/mod.rs

```diff
@@ -99,6 +101,24 @@ impl IndexMetadata {
         &self.index_config().index_uri
     }
 
+    /// Replaces or removes the current retention policy, returning whether a mutation occurred.
+    pub fn set_retention_policy(&mut self, retention_policy_opt: Option<RetentionPolicy>) -> bool {
+        if self.index_config.retention_policy_opt != retention_policy_opt {
+            self.index_config.retention_policy_opt = retention_policy_opt;
+            return true;
+        }
+        false
+    }
```

Comment:
nitpick (as it does not matter much here)

Early exit style goes:

```suggestion
        if self.index_config.retention_policy_opt == retention_policy_opt {
            return false;
        }
        self.index_config.retention_policy_opt = retention_policy_opt;
        true
    }
```

Because the function is so short, the benefit of early exit is very limited... but then I personally prefer having a else statement.


```
 if self.index_config.retention_policy_opt != retention_policy_opt {
     self.index_config.retention_policy_opt = retention_policy_opt;
     true
} else {
     false
}
```